common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -768,6 +768,28 @@ public static void freeDirectNoCleaner(ByteBuffer buffer) {
         decrementMemoryCounter(capacity);
     }
 
+    public static boolean hasAlignDirectByteBuffer() {
+        return hasUnsafe() || PlatformDependent0.hasAlignSliceMethod();
+    }
+
+    public static ByteBuffer alignDirectBuffer(ByteBuffer buffer, int alignment) {
```

Comment:
should we require that `alignment` is a power of two and so we can use `&` as a replacement for `%` for perf reasons ?

Reply:
I switched to using JCTools Pow2 for this, which do these checks.